src/components/wizard/wizard.service.ts

```diff
@@ -1,19 +1,18 @@
 import { Injectable } from '@angular/core';
 import { Subject } from 'rxjs';
-import { MarqueeWizardStepComponent } from './marquee-wizard-step.component';
 
 /**
  * This service is required to provide a form of communication
  * between the marquee wizard steps and the containing marquee wizard.
- * We cannot inject the Host due to the steps being content children 
+ * We cannot inject the Host due to the steps being content children
  * rather than view children.
  */
 @Injectable()
-export class MarqueeWizardService {
-    valid$ = new Subject<MarqueeWizardValidEvent>();
+export class WizardService<TWizardStep> {
+    valid$ = new Subject<WizardValidEvent<TWizardStep>>();
```

Comment:
Since you are making changes here, this should be an EventEmitter and be named something like validChange because it is an event and not a representation of state.